### ISSUE TITLE
Docker-compose down takes 10 seconds

### ISSUE THREAD

**Issue by fsteggink** [geopython/GeoHealthCheck#372]
When using GHC with Docker I've noticed that the command `docker-compose down` takes about 10 seconds. This is an indication that the containers are removed forcefully by Docker. Probably the termination signal sent by Docker is not captured properly, which results in an unclean shutdown.

**Comment by justb4** [comment]
Looks like [this question on StackOverflow](https://stackoverflow.com/questions/43584760/gunicorn-graceful-stopping-with-docker-compose). Related to `gunicorn`. Other sources also suggest to use `CMD exec ...`.
But `gunicorn` itself may also react on HUP signals in later versions: https://docs.gunicorn.org/en/stable/signals.html .

**Comment by fsteggink** [comment]
Yes, that is definitely the case.  I'd use a trap for the TERM signal though, since that is what Docker sends (in my experience). In my own containers I've used signal handlers for the TERM signal and they work well.

**Comment by fsteggink** [comment]
I can confirm that prepending `exec` for the `gunicorn` command works. The ghc_web command is shut down after a few seconds. However, the ghc_runner command still needs 10 seconds to shut down, although I'm using `exec paver` now.

**Comment by fsteggink** [comment]
From the Gitter chat: I've noticed that when starting the runnerwithin the web app, multiple schedulers are started. This is because Gunicorn starts multiple workers. That might lead to potential conflicts (like same check executed simultaneously). Luckily the number of workers can be reduced by setting WSGI_WORKERS=1.

I'll submit a PR as a partial fix for this issue.

**Comment by fsteggink** [comment]
The correct PR for the partial fix is #376.
